src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -14,37 +25,95 @@ interface Props {
   appSettings: {
     isPinSet: boolean;
   };
+  setTimeCounter: (timestamp: number) => SetTimeCounterAction;
+  setFailedAttempts: (attempt: number) => SetFailedAttemptsAction;
+  timeCounter: TimeCounterState;
 }
 
 interface State {
   pin: string;
   error: string;
+  failedTimes: number;
+  isCount: boolean;
 }
 
-export class CurrentPinScreen extends PureComponent<Props, State> {
+class CurrentPinScreen extends PureComponent<Props, State> {
   state = {
     pin: '',
     error: '',
+    failedTimes: 0,
+    isCount: true,
+  };
+
+  handleFailedAttempt = (increasedFailedTimes: number) => {
+    const isFinalAttempt = increasedFailedTimes > 2;
+    if (isFinalAttempt) {
+      const { attempt } = this.props.timeCounter;
+      let timestamp = dayjs();
+
+      if (attempt === 0) {
```

Comment:
so this is duplicated logic with below, you are checking the same else ifs for atttempts, merge those two.
if (attempt === 0) {
      blockedTimeInMinutes = 1;
    } else if (attempt === 1) {
      blockedTimeInMinutes = 2;
    } else if (attempt > 1) {
      blockedTimeInMinutes = 10;
    }
    timestamp = timestamp.add(blockedTimeInMinutes, 'minute');